test/e2e/broker_default_test.go

```diff
@@ -43,135 +42,229 @@ const (
+		deprecatedTriggerFilter bool        //TriggerFilter with DeprecatedSourceAndType or not
+		extensionExists         bool
+	}{{
+		name: "test default broker with many deprecated triggers",
+		eventsToReceive: []eventReceiver{
+			{eventMeta{Type: any, Source: any, ExtensionName: nilString, ExtensionValue: nilString}, newSelector()},
+			{eventMeta{Type: eventType1, Source: any, ExtensionName: nilString, ExtensionValue: nilString}, newSelector()},
+			{eventMeta{Type: any, Source: eventSource1, ExtensionName: nilString, ExtensionValue: nilString}, newSelector()},
+			{eventMeta{Type: eventType1, Source: eventSource1, ExtensionName: nilString, ExtensionValue: nilString}, newSelector()},
+		},
+		eventsToSend: []eventMeta{
+			{eventType1, eventSource1, nilString, nilString},
+			{eventType1, eventSource2, nilString, nilString},
+			{eventType2, eventSource1, nilString, nilString},
+			{eventType2, eventSource2, nilString, nilString},
+		},
+		deprecatedTriggerFilter: true,
+		extensionExists:         false,
+	}, {
+		name: "test default broker with many attribute triggers",
+		eventsToReceive: []eventReceiver{
+			{eventMeta{Type: any, Source: any, ExtensionName: nilString, ExtensionValue: nilString}, newSelector()},
+			{eventMeta{Type: eventType1, Source: any, ExtensionName: nilString, ExtensionValue: nilString}, newSelector()},
+			{eventMeta{Type: any, Source: eventSource1, ExtensionName: nilString, ExtensionValue: nilString}, newSelector()},
+			{eventMeta{Type: eventType1, Source: eventSource1, ExtensionName: nilString, ExtensionValue: nilString}, newSelector()},
+		},
+		eventsToSend: []eventMeta{
+			{eventType1, eventSource1, nilString, nilString},
+			{eventType1, eventSource2, nilString, nilString},
+			{eventType2, eventSource1, nilString, nilString},
+			{eventType2, eventSource2, nilString, nilString},
+		},
+		deprecatedTriggerFilter: false,
+		extensionExists:         false,
+	},
+		{
+			name: "test default broker with many attribute and extension triggers",
+			eventsToReceive: []eventReceiver{
+				{eventMeta{Type: any, Source: any, ExtensionName: extensionName, ExtensionValue: extensionValue}, newSelector()},
+				{eventMeta{Type: eventType1, Source: any, ExtensionName: extensionName, ExtensionValue: extensionValue}, newSelector()},
+				{eventMeta{Type: any, Source: any, ExtensionName: extensionName, ExtensionValue: any}, newSelector()},
+				{eventMeta{Type: any, Source: eventSource1, ExtensionName: extensionName, ExtensionValue: extensionValue}, newSelector()},
+			},
+			eventsToSend: []eventMeta{
+				{eventType1, eventSource1, extensionName, extensionValue},
+				{eventType1, eventSource2, extensionName, extensionValue},
+				{eventType2, eventSource1, extensionName, "non.matching.ext.val"},
+				{eventType2, eventSource2, "non.matching.ext.name", extensionValue},
+			},
+			deprecatedTriggerFilter: false,
+			extensionExists:         true,
+		},
 	}
+	for _, test := range tests {
```

Comment:
I think this won't allow the tests to run in parallel. So it might be better to have different Test* methods that will end up calling some private helper. Does it make sense?